CONTRIBUTING.md

```diff
@@ -129,6 +129,42 @@ Run `rush build -t <packagename>` to build a single project, and all local proje
 
+### Testing
+
+Once the project has successfully built, you can test any of our packages by executing `rushx test` from the root of the package that you want to test. This will automatically run the tests in both NodeJS and in a browser. To target these environments individually, you can run `rushx test:node` and `rushx test:browser`.
+
+By default, these npm scripts run recorded tests. We use a custom recording library called `test-utils-recorder`. You can learn more about our recorder at the [test-utils-recorder's readme](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/test-utils/recorder/README.md).
+
+#### Recorded tests
+
+By default, the npm scripts `test`, `test:node` and `test:browser` run recorded tests in playback mode i.e there is no request made to the service. For http requests made in each test case, there is a recorded response that is used which reproduces the service behavior.
+
+Keep in mind that if you modify any code that is being tested, or any test code, that interacts with the number of network calls that will be made during testing, or the shape of those network calls, the recordings will need to be re-generated.
+
+You can read more in the section: [Regenerating recordings](#regenerating-recordings).
+
+#### Live tests
+
+By default, the npm scripts `test`, `test:node` or `test:browser` run recorded tests. To run live tests, you need the right Azure resources, set appropriate environment variables to point to these resources and set the environment variable `TEST_MODE` to "live". The readme file in the test folder of each package lists the Azure resources you would need and the environment variables to update.
+
+You can create the Azure resources manually on your own or automate this process by...
+
+Our integration tests will run against the live resources, which are determined by the environment variables you provide.
+
+To run the integration tests, you will need to execute `rushx integration-test`.
+
+To generate new resources for your tests, follow the procedure at [Integration Testing with live services](https://github.com/Azure/azure-sdk-for-js/blob/master/CONTRIBUTING.md#integration-testing-with-live-services). What resources get created before the test, or created during the test depends on how the ARM template named `test-resources.json` is built, just as much as how the tests are written.
+
+Integration tests won't need updated recordings, and they also won't change previous recordings unless specified (see [Regenerating recordings](#regenerating-recordings)).
+
+#### Regenerating recordings
+
+If you modify the test files or the underlying network calls of the project you're testing, the recordings need to be re-generated before you run the unit tests.
+
+To re-generate the recordings, set the environment variable `TEST_MODE` to `record` and then run the integration tests. When this process finishes without errors, the recordings will be updated.
```

Comment:
As long as `TEST_MODE` is set to `live`, one can still use the npm script `test` to re-generate the recordings right? If so, then I would not call out "integration" here.